contracts/cw20-staking/src/error.rs

```diff
@@ -0,0 +1,60 @@
+use cosmwasm_std::{StdError, Uint128};
+use thiserror::Error;
+
+#[derive(Error, Debug)]
+pub enum ContractError {
+    #[error("{0}")]
+    Std(#[from] StdError),
+
+    #[error("Unauthorized")]
+    Unauthorized {},
+
+    #[error("Validator '{0}' not in current validator set")]
+    NotInValidatorSet(String),
+
+    #[error("Different denominations in bonds: '{0}' vs. '{1}'")]
+    DifferentBondDenom(String, String),
```

Comment:
Question: would it make sense to have these named (struct) variants rather than tuple variants, where there is more than one argument?

I think the generated code is the same, you just then use `DifferentBondDenom{contract: String, message: String}` or something like that to make it clear which side is which.

Reply:
I think it makes sense, yes. I'll change it for clarity.